lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1393,9 +1411,11 @@ void DifferentiableActivityInfo::analyze(DominanceInfo *di,
         // Handle indirect results in `apply`.
         if (auto *ai = dyn_cast<ApplyInst>(&inst)) {
           auto checkAndSetUseful = [&](SILValue res) {
-            if (isUseful(res, i))
+            if (isUseful(res, i)) {
               for (auto arg : ai->getArgumentsWithoutIndirectResults())
                 setUsefulIfDifferentiable(arg, i);
+              setUsefulIfDifferentiable(ai->getCallee(), i);
```

Comment:
I am not sure if this is necessary for this patch. Looks like it's only needed for full currying.